legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/validator/MappingValidator.java

```diff
@@ -231,7 +235,50 @@ public void validateMappingElementIds(PureModel pureModel, Map<String, Mapping>
         });
     }
 
-    private void validatePropertyMappings(PureModel pureModel, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping mapping, Map<String, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping> mappingByClassMappingId)
+    private void validatePropertyMappings(PureModel pureModel, Map<String, Mapping> mappings, Map<String, org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping> pureMappings)
+    {
+        pureMappings.forEach((mappingPath, mapping) ->
+        {
+            try
+            {
+                mapping._classMappings().forEach(cm ->
+                        {
+                            if(cm instanceof InstanceSetImplementation)
+                            {
+                                ((InstanceSetImplementation)cm)._propertyMappings().forEach(pm ->
+                                        {
+                                            if(pm instanceof Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl && ((Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl) pm)._transform!=null && ((Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl) pm)._transform instanceof Root_meta_pure_metamodel_function_LambdaFunction_Impl)
+                                            {
+
+                                                    Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl ppm = (Root_meta_pure_mapping_modelToModel_PurePropertyMapping_Impl) pm;
+                                                    if(ppm._transform!= null && ppm._transform instanceof Root_meta_pure_metamodel_function_LambdaFunction_Impl){
+                                                        (((Root_meta_pure_metamodel_function_LambdaFunction_Impl)ppm._transform)._expressionSequence).forEach(ex ->
+                                                        {
+                                                            RichIterable<? extends FunctionExpression> fe = core_pure_corefunctions_metaExtension.Root_meta_pure_functions_meta_findExpressionsForFunctionInValueSpecification_ValueSpecification_1__Function_MANY__FunctionExpression_MANY_
+                                                                    ((ValueSpecification)ex,
+                                                                            Lists.mutable.of(pureModel.getFunction("meta::pure::functions::lang::new_Class_1__String_1__KeyExpression_MANY__T_1_",true)),
+                                                                            pureModel.getExecutionSupport());
+                                                            if(fe.size() > 0)
+                                                            {
+                                                                throw new EngineException("Invalid function new_Class_1__String_1__KeyExpression_MANY__T_1_ called " + fe.size() + " times");
```

Comment:
Can we have a more meaningful exception message please e.g. 'The new function/operator(^) is unsupported in property mappings, violated in: "+pm.getId()'